### PR TITLE
Parametrize Binomial and Categorical distributions via logit_p

### PR DESCRIPTION
**Pull request by purna135**
Addressing #5005

Parametrized the Binomial and Categorical distributions via logit_p
Added test in test_distributions_random.py

### CODE REVIEW

**Comment by purna135** [comment]
Hi @ricardoV94 and @MarcoGorelli please have a look.

File: pymc/tests/test_distributions_random.py [hunk 1]
```diff
@@ -1410,6 +1429,28 @@ class TestCategorical(BaseTestDistributionRandom):
     ]
 
 
+class TestLogitCategorical(BaseTestDistributionRandom):
+    pymc_dist = pm.Categorical
+    pymc_dist_params = {"logit_p": np.array([-0.944461608841, 0.489548225319, -2.197224577336])}
```
Comment (ricardoV94) [inline pymc/tests/test_distributions_random.py]: Instead of hardcoding these values, maybe better to just call `scipy.special.softmax` on the original inputs. Same for the logit_p in the Binomial (I think there it would be `expit`)

**Comment by ricardoV94** [comment]
Looks great. Left a small comment above about the tests

**Comment by purna135** [comment]
Thank you @ricardoV94, let me fix that

**Comment by purna135** [comment]
Thank you @ricardoV94, let me fix that

File: pymc/tests/test_distributions_random.py (hunk 1 above)
Comment (purna135) [reply pymc/tests/test_distributions_random.py]: Hi @ricardoV94, Do you mean that it should be as follows?
```
from scipy.special import expit, softmax

class TestLogitCategorical(BaseTestDistributionRandom):
    pymc_dist = pm.Categorical
    pymc_dist_params = {"logit_p": softmax}
    expected_rv_op_params = {"p": np.array([0.28, 0.62, 0.10])}
```

same for Binomial
```
class TestLogitBinomial(BaseTestDistributionRandom):
    pymc_dist = pm.Binomial
    pymc_dist_params = {"n": 100, "logit_p": expit}
    expected_rv_op_params = {"n": 100, "p": 0.9}
```

Comment (ricardoV94) [reply pymc/tests/test_distributions_random.py]: More something like:

 ```python
from scipy.special import expit, softmax

class TestLogitBinomial(BaseTestDistributionRandom):
     pymc_dist = pm.Binomial
     pymc_dist_params = {"n": 100, "logit_p": 0.5}
     expected_rv_op_params = {"n": 100, "p": expit(0.5)}
 ```

Comment (purna135) [reply pymc/tests/test_distributions_random.py]: Yes, got it now
there was a test something like this:

```
class TestBernoulliLogitP(BaseTestDistributionRandom):
    pymc_dist = pm.Bernoulli
    pymc_dist_params = {"logit_p": 1.0}
    expected_rv_op_params = {"p": expit(1.0)}
    checks_to_run = ["check_pymc_params_match_rv_op"]
```

is this same for `softmax` ?

Comment (ricardoV94) [reply pymc/tests/test_distributions_random.py]: Should be similar for softmax yeah, just that it takes a vector... or more. Actually we should test it with a matrix, to cover the fact we are doing `axis=-1`

Comment (purna135) [reply pymc/tests/test_distributions_random.py]: ok, Thanks

Comment (purna135) [reply pymc/tests/test_distributions_random.py]:
```
class TestLogitCategorical(BaseTestDistributionRandom):
    pymc_dist = pm.Categorical
    pymc_dist_params = {"logit_p": np.array([0.28, 0.62, 0.10])}
    expected_rv_op_params = {"p": softmax(np.array([0.28, 0.62, 0.10]))}
```

is this ok now ?

Comment (purna135) [reply pymc/tests/test_distributions_random.py]: And one more question, how do I learn these things?
I'm a complete novice, and I'm having a lot of trouble figuring it out.

Comment (ricardoV94) [reply pymc/tests/test_distributions_random.py]: Should pass a matrix (at least two rows) because of the `axis=-1` setting. 
```python
np.array([[0.28, 0.62, 0.10], [0.28, 0.62, 0.10]])
```

Comment (purna135) [reply pymc/tests/test_distributions_random.py]: can't we do like ?
```
 pymc_dist_params = {"logit_p": np.array([0.28, 0.62, 0.10])}
 expected_rv_op_params = {"p": softmax(np.array([0.28, 0.62, 0.10]), axis=-1)}
```

Comment (ricardoV94) [reply pymc/tests/test_distributions_random.py]: The problem with that is that if you set axis=None, it would give the same result

Comment (purna135) [reply pymc/tests/test_distributions_random.py]: So it should be as bellow, right ?
```

class TestLogitCategorical(BaseTestDistributionRandom):
    pymc_dist = pm.Categorical
    pymc_dist_params = {"logit_p": np.array([[0.28, 0.62, 0.10], [0.28, 0.62, 0.10]])}
    expected_rv_op_params = {"p": softmax(np.array([[0.28, 0.62, 0.10], [0.28, 0.62, 0.10]]))}
```


Please : )
> And one more question, how do I learn these things?
> I'm a complete novice, and I'm having a lot of trouble figuring it out.

Comment (ricardoV94) [reply pymc/tests/test_distributions_random.py]: Should mention axis explicitly again, but it's fine yes
```python
expected_rv_op_params = {"p": softmax(np.array([[0.28, 0.62, 0.10], [0.28, 0.62, 0.10]]), axis=-1)}
```
> And one more question, how do I learn these things?

Nothing like doing what you are doing :)

Comment (purna135) [reply pymc/tests/test_distributions_random.py]: Thank you : )

**Comment by ricardoV94** [comment]
Ah and we need a note in the release notes about the new feature!

**Comment by purna135** [comment]
Okay, I'm not sure what the proper message should be; could you please advise?

**Comment by ricardoV94** [comment]
> Okay, I'm not sure what the proper message should be; could you please advise?

The easiest is to look at previous entries that are similar for inspiration. I spotted these two:

https://github.com/pymc-devs/pymc/blob/436f94a8817e2d1ae0f92b979dfa93d95da5c338/RELEASE-NOTES.md?plain=1#L276

https://github.com/pymc-devs/pymc/blob/436f94a8817e2d1ae0f92b979dfa93d95da5c338/RELEASE-NOTES.md?plain=1#L597

**Comment by purna135** [comment]
I think, I also forgot to add `logit_p` to doc string

**Comment by purna135** [comment]
Done now : )
Please have a look.

**Review by ricardoV94 — APPROVED** [review]

**Comment by ricardoV94** [comment]
Thanks for your help @purna135!

**Comment by purna135** [comment]
Thank you for providing these opportunities @ricardoV94
Could you please assist with #5076 as well?
